src/models/stores/serial.ts

```diff
@@ -0,0 +1,98 @@
+import { NodeChannelInfo } from "src/plugins/dataflow-tool/model/utilities/node";
+
+export class SerialDevice {
+    value: string;
+    localBuffer: string;
+    private port: any; // SERIAL TODO: types SerialPort | null 
+
+    constructor() {
+      this.value = '0';
+      this.localBuffer = '';
+    }
+
+    public hasPort(){
+      return this.port !== undefined;
+    }
+  
+    public async requestAndSetPort(){
+      try {
+          await (navigator as any).serial.requestPort()
+          .then((p: any) => {
+            this.port = p;
+          });
+      }
+
+      catch (error) {
+          console.log('error requesting port: ', error);
```

Comment:
```suggestion
        console.error('error requesting port: ', error);
```
`error` is more appropriate in this context and also doesn't trigger lint warnings.